storage/innobase/srv/srv0start.cc

```diff
@@ -1939,6 +1940,8 @@ dberr_t srv_start(bool create_new_db)
 		return(srv_init_abort(err));
 	}
 
+        innodb_binlog_startup_init();
```

Comment:
There should be some error handling, right?

Reply:
In this particular case, `innodb_binlog_startup_init()` cannot really fail. It is a low-level init that mustly initializes mutexes and such. It does not yet create any binlog files or write to them; this only happens later, when requested from the server layer to open the binlog (and only occurs if the binlog is enabled by the user).

But in general, I'm unsure what to do with certain types of error handling. I've noticed some places in InnoDB code where it does `ut_a()` on error conditions. If we're deep in transaction commit (or even mini-transaction commit) and get an error writing to a log file or other file, what can we do?

Maybe we can discuss this further, I'd like to hear your experiences around how error handling happens currently in InnoDB.

Reply:
I fixed the code to return DB_OUT_OF_MEMORY if it cannot allocate memory needed during initialization.